pennylane/devices/default_clifford.py

```diff
@@ -534,46 +752,233 @@ def _measure_state(self, tableau_simulator, circuit, global_phase):
-            return qml.math.dot(coeffs, expecs)
+        return entropy / qml.math.log(log_base)
 
-        # Add support for more case when the time is right
-        raise NotImplementedError(
-            f"default.clifford doesn't support expectation value calculation with {type(meas_op.obs)} at the moment."
+    def _measure_purity(self, stabilizer, wires):
+        r"""Measure the purity of the state of simulator device
+
+        Computes the state purity using the monotonically decreasing second-order Rényi entropy
+        form given in `Sci Rep 13, 4601 (2023) <https://www.nature.com/articles/s41598-023-31273-9>`_.
+        We utilize the fact that Rényi entropies are independent of the Rényi indices ``n`` for the
+        stabilizer states.
+
+        Args:
+            stabilizer (TensorLike): stabilizer set for the system
+            wires (Iterable): wires describing the subsystem
+            log_base (int): base for the logarithm.
+
+        Returns:
+            (float): entanglement entropy of the subsystem
+        """
+        return 2 ** (-self._measure_stabilizer_vn_entropy(stabilizer, wires, log_base=2))
+
+    @property
+    def probability_target(self):
+        """Get the target computational basis states for computing outcome probability."""
+        return self._prob_states
+
+    @probability_target.setter
+    def probability_target(self, basis_states):
+        """Set the target computational basis states for computing outcome probability."""
+        self._prob_states = qml.math.stack(
+            basis_states if len(qml.math.shape(basis_states)) > 1 else [basis_states]
         )
+
+    def _measure_probability(self, tableau_simulator, circuit, meas_op, stim):
+        """Measure the probability of each computational basis state."""
+
+        if self._prob_states is None and self._tableau:
+            raise ValueError(
+                "In order to maintain computational efficiency,\
+                with ``tableau=True``, the clifford device supports returning\
+                probability only for selected target computational basis states.\
+                Please use `probability_target()` method to set them."
+            )
+
+        # we might be able to skip the inverse done below
+        # as the distribution for the inverted tableau should remain same
+        diagonalizing_cit = tableau_simulator.current_inverse_tableau().inverse().to_circuit()
+        diagonalizing_ops = [] if not meas_op.obs else meas_op.obs.diagonalizing_gates()
+        for ops in diagonalizing_ops:
+            # Check if it is Clifford
+            if ops.name not in _GATE_OPERATIONS:  # pragma: no cover
+                raise ValueError(
+                    f"Currently, we only support observables whose diagonalizing gates are Clifford, got {ops}"
+                )
+            # Add to the circuit to rotate the basis
+            stim_op = self.pl_to_stim(ops)
```

Comment:
nitpick (unless you're using it elsewhere?)
```suggestion
        for op in diagonalizing_ops:
            # Check if it is Clifford
            if op.name not in _GATE_OPERATIONS:  # pragma: no cover
                raise ValueError(
                    f"Currently, we only support observables whose diagonalizing gates are Clifford, got {op}"
                )
            # Add to the circuit to rotate the basis
            stim_op = self.pl_to_stim(op)
```